### PR TITLE
Add prow job config for pushing cloud provider azure image to gcr staing repo.

### PR DESCRIPTION
**Pull request by nilo19**
Add prow job config for pushing cloud provider azure image to gcr staing repo.

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-provider-azure.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+postsubmits:
+  # This is the github repo we'll build from. This block needs to be repeated
+  # for each repo.
+  kubernetes-sigs/some-repo-name:
```
Comment (feiskyer) [inline config/jobs/image-pushing/k8s-staging-provider-azure.yaml]: kubernetes-sigs/cloud-provider-azure

Comment (nilo19) [reply config/jobs/image-pushing/k8s-staging-provider-azure.yaml]: oops

File: config/jobs/image-pushing/k8s-staging-provider-azure.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  # This is the github repo we'll build from. This block needs to be repeated
+  # for each repo.
+  kubernetes-sigs/cloud-provider-azure:
+    # The name should be changed to match the repo name above
+    - name: post-provider-azure-push-images
+      cluster: test-infra-trusted
+      annotations:
+        # This is the name of some testgrid dashboard to report to.
+        # If this is the first one for your sig, you may need to create one
+        testgrid-dashboards: sig-cloud-provider-image-pushes
```
Comment (feiskyer) [inline config/jobs/image-pushing/k8s-staging-provider-azure.yaml]: Refer here https://github.com/kubernetes/test-infra/blob/9bcc9e7640e70cbcaa40714b88a8cf4a1e3f4d29/config/testgrids/kubernetes/sig-cloud-provider/azure/config.yaml#L1 to add a dashboard

**Comment by BenTheElder** [comment]
not overly familiar with the new image push tooling but this looks about right
/lgtm
/approve

**Pull request merged**